File: myExpenses/src/main/java/org/totschnig/myexpenses/model/ExportFormat.java
package org.totschnig.myexpenses.model;

import java.util.Locale;

public enum ExportFormat {
  QIF {
    @Override
    public String getMimeType() {
      return "application/qif";
    }
  }, CSV {
    @Override
    public String getMimeType() {
      return "text/csv";
    }
  };

  public abstract String getMimeType();

  public String getExtension() {
    return name().toLowerCase(Locale.US);
  }
}

File: myExpenses/src/main/java/org/totschnig/myexpenses/model/ExportFormat.kt
package org.totschnig.myexpenses.model

import org.totschnig.myexpenses.R
import java.util.*

enum class ExportFormat(val mimeType: String, val resId: Int) {
    QIF("application/qif", R.id.qif),
    CSV("text/csv", R.id.csv),
    JSON("application/json", R.id.json);

    val extension: String
        get() = name.lowercase(Locale.US)
}